tokio-stream/src/stream_ext.rs

```diff
@@ -993,6 +995,45 @@ pub trait StreamExt: Stream {
     {
         throttle(duration, self)
     }
+
+    /// Collects items into batches inside a vector within a deadline.
+    ///
+    /// `chunks_timeout` attempts to yield a vector of len `capacity` within the deadline,
+    /// otherwise will yield a vector of len less than `capacity` if the deadline is reached
```

Comment:
Can you improve the documentation here? In general, I would avoid sentences that start with a piece of code, and its also missing a period. Beyond that, a more detailed explanation would be nice.